### PR TITLE
Rule S5757: Allowing confidential information to be logged is security-sensitive

### LINKED ISSUE #1965: RSPEC-5757 Allowing confidential information to be logged is security-sensitive
Implements [RSPEC-5757](https://jira.sonarsource.com/browse/RSPEC-5757)

For both JS/TS

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-5757

Report on `secrets: []` as main location and `new Signale` as secondary location:
```
const options = {
  secrets: []  // Sensitive main location
};
 
const logger = new Signale(options); // Sensitive secondary location
```

If `options` argument of `new Signale(options)` call doesn't contain `secret` property or `options` argument is not defined, the primary location should be on `new Signale()`:
```
const options = {
};
 
const logger1 = new Signale(options); // Sensitive
const logger2 = new Signale(); // Sensitive
```

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
Fixes #1965

### CODE REVIEW

File: eslint-bridge/tests/rules/confidential-information-logging.test.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ruleTester.run('Allowing confidential information to be logged is security-sensitive', rule, {
+  valid: [
+    {
+      code: `
+      const { Signale } = require('signale');
+      const options = {
+        secrets: ["([0-9]{4}-?)+"]
+      };
+      const logger = new Signale(options); // Compliant
+      const CREDIT_CARD_NUMBERS = ["1234-5678-0000-9999", "1234-5678-0000-8888"];
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/tests/rules/confidential-information-logging.test.ts:35]: do you think we need this extra code? I think it's not relevant for the unit test itself

File: eslint-bridge/tests/rules/confidential-information-logging.test.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  invalid: [
+    {
+      code: `
+      const signale = require('signale');
+      const CREDIT_CARD_NUMBERS = ["1234-5678-0000-9999", "1234-5678-0000-8888"];
+      const options = {
+        secrets: []
+      };
+      const logger = new signale.Signale(options); // Sensitive
+      CREDIT_CARD_NUMBERS.forEach(function(CREDIT_CARD_NUMBER) {
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/tests/rules/confidential-information-logging.test.ts:94]: same here..I don't think we need the code handling `CREDIT_CARD_NUMBERS`

**Review by andrea-guarino-sonarsource — APPROVED** [review]
LGTM, some minor comments about unit tests

File: eslint-bridge/tests/rules/confidential-information-logging.test.ts (hunk 1 above)
Comment (guillaume-dequenne-sonarsource) [reply eslint-bridge/tests/rules/confidential-information-logging.test.ts:35]: Indeed, I'll simplify those tests a bit.

**Pull request merged**